DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -871,17 +980,23 @@ public DMASTStatement Statement(bool requireDelimiter = true)
                 Check(TokenType.DM_Colon);
                 Whitespace();
 
+                Token ifBody = Current();
                 DMASTProcStatement procStatement = ProcStatement();
                 DMASTProcBlockInner body;
                 DMASTProcBlockInner elseBody = null;
 
                 if (procStatement != null) {
-                    body = new DMASTProcBlockInner(loc, new DMASTProcStatement[] { procStatement });
+                    if (procStatement.IsAggregateOr<DMASTProcStatementSet>()) // Having just a set statement counts as being empty.
+                        Error(WarningCode.EmptyBlock, "Empty if-block detected");
```

Comment:
I think it'd be better to check the body's statements inside `DMProcBuilder.ProcessStatementIf()` instead of these several different checks during AST generation. Maybe even `DMProcBuilder.ProcessBlockInner()` if you want to more comprehensively catch every empty block.